lib/spack/spack/cmd/config.py

```diff
@@ -275,12 +292,164 @@ def config_remove(args):
+    for scope in updates:
+        cfg_file = spack.config.config.get_config_filename(
+            scope.name, args.section
+        )
+        with open(cfg_file) as f:
+            data = syaml.load(f) or {}
+            data = data.pop(args.section, {})
+        update_fn(data)
+
+        # Make a backup copy and rewrite the file
+        bkp_file = cfg_file + '.bkp'
+        shutil.copy(cfg_file, bkp_file)
+        spack.config.config.update_config(
+            args.section, data, scope=scope.name, force=True
+        )
+        msg = 'File "{0}" updated [backup={1}]'
+        tty.msg(msg.format(cfg_file, bkp_file))
+
+
+def _can_revert_update(scope_dir, cfg_file, bkp_file):
+    dir_ok = fs.can_write_to_dir(scope_dir)
+    cfg_ok = not os.path.exists(cfg_file) or fs.can_access(cfg_file)
+    bkp_ok = fs.can_access(bkp_file)
+    return dir_ok and cfg_ok and bkp_ok
+
+
+def config_revert(args):
+    scopes = [args.scope] if args.scope else [
+        x.name for x in spack.config.config.file_scopes
+    ]
+
+    # Search for backup files in the configuration scopes
+    Entry = collections.namedtuple('Entry', ['scope', 'cfg', 'bkp'])
+    to_be_restored, cannot_overwrite = [], []
+    for scope in scopes:
+        cfg_file = spack.config.config.get_config_filename(scope, args.section)
+        bkp_file = cfg_file + '.bkp'
+
+        # If the backup files doesn't exist move to the next scope
+        if not os.path.exists(bkp_file):
+            continue
+
+        # If it exists and we don't have write access in this scope
+        # keep track of it and report a comprehensive error later
+        entry = Entry(scope, cfg_file, bkp_file)
+        scope_dir = os.path.dirname(bkp_file)
+        can_be_reverted = _can_revert_update(scope_dir, cfg_file, bkp_file)
+        if not can_be_reverted:
+            cannot_overwrite.append(entry)
+            continue
+
+        to_be_restored.append(entry)
+
+    # Report errors if we can't revert a configuration
+    if cannot_overwrite:
+        msg = 'Detected permission issues with the following scopes:\n\n'
+        for e in cannot_overwrite:
+            msg += '\t[scope={0.scope}, cfg={0.cfg}, bkp={0.bkp}]\n'.format(e)
+        msg += ('\nEither ensure to have the right permissions before retrying'
+                ' or be more specific on the scope to revert.')
+        tty.die(msg)
+
+    proceed = True
+    if not args.yes_to_all:
+        msg = ('The following scopes will be restored from the corresponding'
+               ' backup files:\n')
+        for entry in to_be_restored:
+            msg += '\t[scope={0.scope}, bkp={0.bkp}]\n'.format(entry)
+        msg += 'This operation cannot be undone.'
+        tty.msg(msg)
+        proceed = tty.get_yes_or_no('Do you want to proceed?', default=False)
+
+    if not proceed:
+        tty.die('Operation aborted.')
+
+    for _, cfg_file, bkp_file in to_be_restored:
+        shutil.copy(bkp_file, cfg_file)
+        os.unlink(bkp_file)
```

Comment:
Should we replace the backup with the config file?

Reply:
I would start with this simple mechanism and eventually complicate it as we receive feedback from users on `develop`. Did you try this branch with any of your configuration / environments?

Reply:
Nope, haven't tried this branch yet, just spot checking things.